src/libsystemd-network/dhcp-option.c

```diff
@@ -284,6 +287,102 @@ int dhcp_option_append(DHCPMessage *message, size_t size, size_t *offset,
+                                /* close the options array */
+                                r = option_append(message->options, size, offset,
+                                                  SD_DHCP_OPTION_END, 0, NULL);
+                                if (r < 0)
+                                        return r;
+                        }
+
+                        continue;
+                }
+
+                size_t file_offset = *offset - size;
+                if (use_file && file_offset + 3 < sizeof(message->file)) {
+                        /* still space in the file array for at least TL and end */
+                        available = sizeof(message->file) - 3 - file_offset;
+                        to_write = MIN(available, MIN(optlen, (size_t)UINT8_MAX));
+
+                        r = option_append(message->file, sizeof(message->file), &file_offset,
+                                          code, to_write, optval);
+                        if (r < 0)
+                                return r;
+
+                        optval += to_write;
+                        optlen -= to_write;
+                        *offset = size + file_offset;
+                        if (to_write == available) {
+                                /* close the file array */
+                                r = option_append(message->file, sizeof(message->file), &file_offset,
+                                                  SD_DHCP_OPTION_END, 0, NULL);
+                                if (r < 0)
+                                        return r;
+                        }
+
+                        continue;
+                }
+
+                size_t sname_offset = *offset - size - (use_file ? sizeof(message->file) : 0);
+                if (use_sname && sname_offset + 3 < sizeof(message->sname)) {
+                        /* still space in the sname array for at least TL and end */
+                        available = sizeof(message->sname) - 3 - sname_offset;
+                        to_write = MIN(available, MIN(optlen, (size_t)UINT8_MAX));
+
+                        r = option_append(message->sname, sizeof(message->sname), &sname_offset,
+                                          code, to_write, optval);
+                        if (r < 0)
+                                return r;
+                        optval += to_write;
+                        optlen -= to_write;
+                        *offset = size + (use_file ? sizeof(message->file) : 0) + sname_offset;
+                }
+
+                return -ENOBUFS;
+        }
+
+        return 0;
+}
+
+int dhcp_option_append(DHCPMessage *message, size_t size, size_t *offset,
+                       uint8_t overload,
+                       uint8_t code, size_t optlen, const void *optval) {
+        if (dhcp_option_can_merge(code))
+                return dhcp_option_append_long(message, size, offset, overload, code, optlen, optval);
+        else
```

Comment:
maybe drop the `else` word. it's redundant, and we like code that avoids redundant words

Reply:
Huh too used to Rust's if-is-an-expression now. Done.